### PR TITLE
Changed some int parameters to vector parameters

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

File: src/Entities/ExpBottleEntity.h [hunk 1]
```diff
@@ -30,15 +30,17 @@ class cExpBottleEntity :
 
 	cExpBottleEntity(cEntity * a_Creator, double a_X, double a_Y, double a_Z, const Vector3d & a_Speed);
 
+	cExpBottleEntity(cEntity * a_Creator, Vector3d a_Pos, const Vector3d & a_Speed);
```
Comment (peterbell10) [inline src/Entities/ExpBottleEntity.h]: `a_Speed` could be by value as well.

File: src/World.h [hunk 2]
```diff
@@ -454,7 +454,7 @@ class cWorld :
 
 	/** Spawns a boat at the given coordinates.
 	Returns the UniqueID of the spawned boat, or cEntity::INVALID_ID on failure. */
-	UInt32 SpawnBoat(double a_X, double a_Y, double a_Z, cBoat::eMaterial a_Material);
+	UInt32 SpawnBoat(Vector3d a_Pos, cBoat::eMaterial a_Material);
```
Comment (peterbell10) [inline src/World.h]: This is exported to lua, you'll either need to keep both as overloads or add a manual binding that accepts both forms.

File: src/World.h [hunk 3]
```diff
@@ -463,7 +463,7 @@ class cWorld :
 	/** Spawns a new primed TNT entity at the specified block coords and specified fuse duration.
 	Initial velocity is given based on the relative coefficient provided.
 	Returns the UniqueID of the created entity, or cEntity::INVALID_ID on failure. */
-	UInt32 SpawnPrimedTNT(double a_X, double a_Y, double a_Z, int a_FuseTimeInSec = 80, double a_InitialVelocityCoeff = 1);
+	UInt32 SpawnPrimedTNT(Vector3d a_Pos, int a_FuseTimeInSec = 80, double a_InitialVelocityCoeff = 1);
```
Comment (peterbell10) [inline src/World.h]: Here too.

**Comment by madmaxoft** [comment]
Oh, yes, and I skipped the redstone simulator altogether again. That is @tigerw 's domain :)

**Pull request merged**